src/common/buffer.cc

```diff
@@ -1068,7 +1068,9 @@ static simple_spinlock_t buffer_debug_lock = SIMPLE_SPINLOCK_INITIALIZER;
   template<bool is_const>
   void buffer::list::iterator_impl<is_const>::advance(unsigned o)
   {
-    p_off += o;
+    if (__builtin_add_overflow(p_off, o, &p_off)) {
```

Comment:
may be we can use NDEBUG or something alike to enable debug-only things. like in many other projects. but i checked the definition of `assert()`, seems we don't have such a thing yet.